yarn-project/p2p/src/service/libp2p_service.ts

```diff
@@ -217,6 +224,40 @@ export class LibP2PService implements P2PService {
     this.knownTxLookup.handleSettledTxs(txHashes.map(x => x.toString()));
   }
 
+  private async addPeer(enr: ENR) {
+    const peerMultiAddr = await enr.getFullMultiaddr('tcp');
+    if (!peerMultiAddr) {
+      // No TCP address, can't connect
+      return;
+    }
+    const peerIdStr = peerMultiAddr.getPeerId();
+
+    if (!peerIdStr) {
+      this.logger.info(`Peer ID not found in discovered node's multiaddr: ${peerMultiAddr}`);
+      return;
+    }
+
+    // check if peer is already known
+    const peerId = peerIdFromString(peerIdStr);
+    const hasPeer = await this.node.peerStore.has(peerId);
+
+    // add to peer store if not already known
+    if (!hasPeer) {
+      this.logger.info(`Discovered peer ${enr.peerId().toString()}. Adding to libp2p peer list`);
+      try {
+        const stream = await this.node.dialProtocol(peerMultiAddr, this.protocolId);
+
+        // dial successful, add to DB as well
+        if (!this.peerStore.getPeer(peerIdStr)) {
+          await this.peerStore.addPeer(peerIdStr, enr);
+        }
+        await stream.close();
```

Comment:
I think we should refactor this logic slightly. Currently, if adding to the store fails it will print 'Failed to dial....' and leave the stream open.

Reply:
Shouldn't we remove the `stream.close()` from here?